File: ruby/models/instrument.rb
class Instrument
  attr_accessor :symbol, :state, :typ, :listing, :expiry, :underlying, :buy_leg, :sell_leg, :quote_currency, :reference, :reference_symbol, :tick_size, :multiplier, :settl_currency, :init_margin, :maint_margin, :limit, :opening_timestamp, :closing_timestamp, :prev_close_price, :limit_down_price, :limit_up_price, :volume, :vwap, :high_price, :low_price, :last_price, :last_tick_direction, :last_change_pcnt, :bid_price, :mid_price, :ask_price, :open_interest, :settled_price, :timestamp

  # :internal => :external
  def self.attribute_map
    {
      :symbol => :symbol,
      :state => :state,
      :typ => :typ,
      :listing => :listing,
      :expiry => :expiry,
      :underlying => :underlying,
      :buy_leg => :buyLeg,
      :sell_leg => :sellLeg,
      :quote_currency => :quoteCurrency,
      :reference => :reference,
      :reference_symbol => :referenceSymbol,
      :tick_size => :tickSize,
      :multiplier => :multiplier,
      :settl_currency => :settlCurrency,
      :init_margin => :initMargin,
      :maint_margin => :maintMargin,
      :limit => :limit,
      :opening_timestamp => :openingTimestamp,
      :closing_timestamp => :closingTimestamp,
      :prev_close_price => :prevClosePrice,
      :limit_down_price => :limitDownPrice,
      :limit_up_price => :limitUpPrice,
      :volume => :volume,
      :vwap => :vwap,
      :high_price => :highPrice,
      :low_price => :lowPrice,
      :last_price => :lastPrice,
      :last_tick_direction => :lastTickDirection,
      :last_change_pcnt => :lastChangePcnt,
      :bid_price => :bidPrice,
      :mid_price => :midPrice,
      :ask_price => :askPrice,
      :open_interest => :openInterest,
      :settled_price => :settledPrice,
      :timestamp => :timestamp

    }
  end

  def initialize(attributes = {})
    return if attributes.empty?
    # Morph attribute keys into undescored rubyish style
    if self.class.attribute_map[:"symbol"]
      @symbol = attributes["symbol"]
    end
    if self.class.attribute_map[:"state"]
      @state = attributes["state"]
    end
    if self.class.attribute_map[:"typ"]
      @typ = attributes["typ"]
    end
    if self.class.attribute_map[:"listing"]
      @listing = attributes["listing"]
    end
    if self.class.attribute_map[:"expiry"]
      @expiry = attributes["expiry"]
    end
    if self.class.attribute_map[:"underlying"]
      @underlying = attributes["underlying"]
    end
    if self.class.attribute_map[:"buy_leg"]
      @buy_leg = attributes["buyLeg"]
    end
    if self.class.attribute_map[:"sell_leg"]
      @sell_leg = attributes["sellLeg"]
    end
    if self.class.attribute_map[:"quote_currency"]
      @quote_currency = attributes["quoteCurrency"]
    end
    if self.class.attribute_map[:"reference"]
      @reference = attributes["reference"]
    end
    if self.class.attribute_map[:"reference_symbol"]
      @reference_symbol = attributes["referenceSymbol"]
    end
    if self.class.attribute_map[:"tick_size"]
      @tick_size = attributes["tickSize"]
    end
    if self.class.attribute_map[:"multiplier"]
      @multiplier = attributes["multiplier"]
    end
    if self.class.attribute_map[:"settl_currency"]
      @settl_currency = attributes["settlCurrency"]
    end
    if self.class.attribute_map[:"init_margin"]
      @init_margin = attributes["initMargin"]
    end
    if self.class.attribute_map[:"maint_margin"]
      @maint_margin = attributes["maintMargin"]
    end
    if self.class.attribute_map[:"limit"]
      @limit = attributes["limit"]
    end
    if self.class.attribute_map[:"opening_timestamp"]
      @opening_timestamp = attributes["openingTimestamp"]
    end
    if self.class.attribute_map[:"closing_timestamp"]
      @closing_timestamp = attributes["closingTimestamp"]
    end
    if self.class.attribute_map[:"prev_close_price"]
      @prev_close_price = attributes["prevClosePrice"]
    end
    if self.class.attribute_map[:"limit_down_price"]
      @limit_down_price = attributes["limitDownPrice"]
    end
    if self.class.attribute_map[:"limit_up_price"]
      @limit_up_price = attributes["limitUpPrice"]
    end
    if self.class.attribute_map[:"volume"]
      @volume = attributes["volume"]
    end
    if self.class.attribute_map[:"vwap"]
      @vwap = attributes["vwap"]
    end
    if self.class.attribute_map[:"high_price"]
      @high_price = attributes["highPrice"]
    end
    if self.class.attribute_map[:"low_price"]
      @low_price = attributes["lowPrice"]
    end
    if self.class.attribute_map[:"last_price"]
      @last_price = attributes["lastPrice"]
    end
    if self.class.attribute_map[:"last_tick_direction"]
      @last_tick_direction = attributes["lastTickDirection"]
    end
    if self.class.attribute_map[:"last_change_pcnt"]
      @last_change_pcnt = attributes["lastChangePcnt"]
    end
    if self.class.attribute_map[:"bid_price"]
      @bid_price = attributes["bidPrice"]
    end
    if self.class.attribute_map[:"mid_price"]
      @mid_price = attributes["midPrice"]
    end
    if self.class.attribute_map[:"ask_price"]
      @ask_price = attributes["askPrice"]
    end
    if self.class.attribute_map[:"open_interest"]
      @open_interest = attributes["openInterest"]
    end
    if self.class.attribute_map[:"settled_price"]
      @settled_price = attributes["settledPrice"]
    end
    if self.class.attribute_map[:"timestamp"]
      @timestamp = attributes["timestamp"]
    end
    

  end

  def to_body
    body = {}
    self.class.attribute_map.each_pair do |key, value|
      body[value] = self.send(key) unless self.send(key).nil?
    end
    body
  end
end

File: clients/java/android/main/java/com/wordnik/client/model/instrument.java
package com.wordnik.client.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.Date;
public class Instrument {
  @JsonProperty("symbol")
  private String symbol = null;
  @JsonProperty("state")
  private String state = null;
  @JsonProperty("typ")
  private String typ = null;
  @JsonProperty("listing")
  private Date listing = null;
  @JsonProperty("expiry")
  private Date expiry = null;
  @JsonProperty("underlying")
  private String underlying = null;
  @JsonProperty("buyLeg")
  private String buyLeg = null;
  @JsonProperty("sellLeg")
  private String sellLeg = null;
  @JsonProperty("quoteCurrency")
  private String quoteCurrency = null;
  @JsonProperty("reference")
  private String reference = null;
  @JsonProperty("referenceSymbol")
  private String referenceSymbol = null;
  @JsonProperty("tickSize")
  private Double tickSize = null;
  @JsonProperty("multiplier")
  private Double multiplier = null;
  @JsonProperty("settlCurrency")
  private String settlCurrency = null;
  @JsonProperty("initMargin")
  private Double initMargin = null;
  @JsonProperty("maintMargin")
  private Double maintMargin = null;
  @JsonProperty("limit")
  private Double limit = null;
  @JsonProperty("openingTimestamp")
  private Date openingTimestamp = null;
  @JsonProperty("closingTimestamp")
  private Date closingTimestamp = null;
  @JsonProperty("prevClosePrice")
  private Double prevClosePrice = null;
  @JsonProperty("limitDownPrice")
  private Double limitDownPrice = null;
  @JsonProperty("limitUpPrice")
  private Double limitUpPrice = null;
  @JsonProperty("volume")
  private Double volume = null;
  @JsonProperty("vwap")
  private Double vwap = null;
  @JsonProperty("highPrice")
  private Double highPrice = null;
  @JsonProperty("lowPrice")
  private Double lowPrice = null;
  @JsonProperty("lastPrice")
  private Double lastPrice = null;
  @JsonProperty("lastTickDirection")
  private String lastTickDirection = null;
  @JsonProperty("lastChangePcnt")
  private Double lastChangePcnt = null;
  @JsonProperty("bidPrice")
  private Double bidPrice = null;
  @JsonProperty("midPrice")
  private Double midPrice = null;
  @JsonProperty("askPrice")
  private Double askPrice = null;
  @JsonProperty("openInterest")
  private Double openInterest = null;
  @JsonProperty("settledPrice")
  private Double settledPrice = null;
  @JsonProperty("timestamp")
  private Date timestamp = null;
  public String getSymbol() {
    return symbol;
  }
  public void setSymbol(String symbol) {
    this.symbol = symbol;
  }

  public String getState() {
    return state;
  }
  public void setState(String state) {
    this.state = state;
  }

  public String getTyp() {
    return typ;
  }
  public void setTyp(String typ) {
    this.typ = typ;
  }

  public Date getListing() {
    return listing;
  }
  public void setListing(Date listing) {
    this.listing = listing;
  }

  public Date getExpiry() {
    return expiry;
  }
  public void setExpiry(Date expiry) {
    this.expiry = expiry;
  }

  public String getUnderlying() {
    return underlying;
  }
  public void setUnderlying(String underlying) {
    this.underlying = underlying;
  }

  public String getBuyLeg() {
    return buyLeg;
  }
  public void setBuyLeg(String buyLeg) {
    this.buyLeg = buyLeg;
  }

  public String getSellLeg() {
    return sellLeg;
  }
  public void setSellLeg(String sellLeg) {
    this.sellLeg = sellLeg;
  }

  public String getQuoteCurrency() {
    return quoteCurrency;
  }
  public void setQuoteCurrency(String quoteCurrency) {
    this.quoteCurrency = quoteCurrency;
  }

  public String getReference() {
    return reference;
  }
  public void setReference(String reference) {
    this.reference = reference;
  }

  public String getReferenceSymbol() {
    return referenceSymbol;
  }
  public void setReferenceSymbol(String referenceSymbol) {
    this.referenceSymbol = referenceSymbol;
  }

  public Double getTickSize() {
    return tickSize;
  }
  public void setTickSize(Double tickSize) {
    this.tickSize = tickSize;
  }

  public Double getMultiplier() {
    return multiplier;
  }
  public void setMultiplier(Double multiplier) {
    this.multiplier = multiplier;
  }

  public String getSettlCurrency() {
    return settlCurrency;
  }
  public void setSettlCurrency(String settlCurrency) {
    this.settlCurrency = settlCurrency;
  }

  public Double getInitMargin() {
    return initMargin;
  }
  public void setInitMargin(Double initMargin) {
    this.initMargin = initMargin;
  }

  public Double getMaintMargin() {
    return maintMargin;
  }
  public void setMaintMargin(Double maintMargin) {
    this.maintMargin = maintMargin;
  }

  public Double getLimit() {
    return limit;
  }
  public void setLimit(Double limit) {
    this.limit = limit;
  }

  public Date getOpeningTimestamp() {
    return openingTimestamp;
  }
  public void setOpeningTimestamp(Date openingTimestamp) {
    this.openingTimestamp = openingTimestamp;
  }

  public Date getClosingTimestamp() {
    return closingTimestamp;
  }
  public void setClosingTimestamp(Date closingTimestamp) {
    this.closingTimestamp = closingTimestamp;
  }

  public Double getPrevClosePrice() {
    return prevClosePrice;
  }
  public void setPrevClosePrice(Double prevClosePrice) {
    this.prevClosePrice = prevClosePrice;
  }

  public Double getLimitDownPrice() {
    return limitDownPrice;
  }
  public void setLimitDownPrice(Double limitDownPrice) {
    this.limitDownPrice = limitDownPrice;
  }

  public Double getLimitUpPrice() {
    return limitUpPrice;
  }
  public void setLimitUpPrice(Double limitUpPrice) {
    this.limitUpPrice = limitUpPrice;
  }

  public Double getVolume() {
    return volume;
  }
  public void setVolume(Double volume) {
    this.volume = volume;
  }

  public Double getVwap() {
    return vwap;
  }
  public void setVwap(Double vwap) {
    this.vwap = vwap;
  }

  public Double getHighPrice() {
    return highPrice;
  }
  public void setHighPrice(Double highPrice) {
    this.highPrice = highPrice;
  }

  public Double getLowPrice() {
    return lowPrice;
  }
  public void setLowPrice(Double lowPrice) {
    this.lowPrice = lowPrice;
  }

  public Double getLastPrice() {
    return lastPrice;
  }
  public void setLastPrice(Double lastPrice) {
    this.lastPrice = lastPrice;
  }

  public String getLastTickDirection() {
    return lastTickDirection;
  }
  public void setLastTickDirection(String lastTickDirection) {
    this.lastTickDirection = lastTickDirection;
  }

  public Double getLastChangePcnt() {
    return lastChangePcnt;
  }
  public void setLastChangePcnt(Double lastChangePcnt) {
    this.lastChangePcnt = lastChangePcnt;
  }

  public Double getBidPrice() {
    return bidPrice;
  }
  public void setBidPrice(Double bidPrice) {
    this.bidPrice = bidPrice;
  }

  public Double getMidPrice() {
    return midPrice;
  }
  public void setMidPrice(Double midPrice) {
    this.midPrice = midPrice;
  }

  public Double getAskPrice() {
    return askPrice;
  }
  public void setAskPrice(Double askPrice) {
    this.askPrice = askPrice;
  }

  public Double getOpenInterest() {
    return openInterest;
  }
  public void setOpenInterest(Double openInterest) {
    this.openInterest = openInterest;
  }

  public Double getSettledPrice() {
    return settledPrice;
  }
  public void setSettledPrice(Double settledPrice) {
    this.settledPrice = settledPrice;
  }

  public Date getTimestamp() {
    return timestamp;
  }
  public void setTimestamp(Date timestamp) {
    this.timestamp = timestamp;
  }

  @Override
  public String toString()  {
    StringBuilder sb = new StringBuilder();
    sb.append("class Instrument {\n");
    sb.append("  symbol: ").append(symbol).append("\n");
    sb.append("  state: ").append(state).append("\n");
    sb.append("  typ: ").append(typ).append("\n");
    sb.append("  listing: ").append(listing).append("\n");
    sb.append("  expiry: ").append(expiry).append("\n");
    sb.append("  underlying: ").append(underlying).append("\n");
    sb.append("  buyLeg: ").append(buyLeg).append("\n");
    sb.append("  sellLeg: ").append(sellLeg).append("\n");
    sb.append("  quoteCurrency: ").append(quoteCurrency).append("\n");
    sb.append("  reference: ").append(reference).append("\n");
    sb.append("  referenceSymbol: ").append(referenceSymbol).append("\n");
    sb.append("  tickSize: ").append(tickSize).append("\n");
    sb.append("  multiplier: ").append(multiplier).append("\n");
    sb.append("  settlCurrency: ").append(settlCurrency).append("\n");
    sb.append("  initMargin: ").append(initMargin).append("\n");
    sb.append("  maintMargin: ").append(maintMargin).append("\n");
    sb.append("  limit: ").append(limit).append("\n");
    sb.append("  openingTimestamp: ").append(openingTimestamp).append("\n");
    sb.append("  closingTimestamp: ").append(closingTimestamp).append("\n");
    sb.append("  prevClosePrice: ").append(prevClosePrice).append("\n");
    sb.append("  limitDownPrice: ").append(limitDownPrice).append("\n");
    sb.append("  limitUpPrice: ").append(limitUpPrice).append("\n");
    sb.append("  volume: ").append(volume).append("\n");
    sb.append("  vwap: ").append(vwap).append("\n");
    sb.append("  highPrice: ").append(highPrice).append("\n");
    sb.append("  lowPrice: ").append(lowPrice).append("\n");
    sb.append("  lastPrice: ").append(lastPrice).append("\n");
    sb.append("  lastTickDirection: ").append(lastTickDirection).append("\n");
    sb.append("  lastChangePcnt: ").append(lastChangePcnt).append("\n");
    sb.append("  bidPrice: ").append(bidPrice).append("\n");
    sb.append("  midPrice: ").append(midPrice).append("\n");
    sb.append("  askPrice: ").append(askPrice).append("\n");
    sb.append("  openInterest: ").append(openInterest).append("\n");
    sb.append("  settledPrice: ").append(settledPrice).append("\n");
    sb.append("  timestamp: ").append(timestamp).append("\n");
    sb.append("}\n");
    return sb.toString();
  }
}

